stdlib/Artifacts/src/Artifacts.jl

```diff
@@ -659,10 +659,10 @@ macro artifact_str(name, platform=nothing, artifacts_toml_path=nothing)
         srcfile = pwd()
     end
     # Sometimes we know the exact path to the Artifacts.toml file, so we can save some lookups
-    local artifacts_toml = if artifacts_toml_path === nothing
+    local artifacts_toml = if artifacts_toml_path === nothing || artifacts_toml_path == :(nothing)
         find_artifacts_toml(srcfile)
     else
-        artifacts_toml_path
+        eval(artifacts_toml_path)
```

Comment:
Did you try this @staticfloat? It seems to error for me:

```
In [3]: using Vulkan_Headers_jll
[ Info: Precompiling Vulkan_Headers_jll [8d446b21-f3ad-5576-a034-752265b9b6f9]
ERROR: LoadError: Evaluation into the closed module `Artifacts` breaks incremental compilation because the side effects will not be permanent. This is likely due to some other module mutating `Artifacts` with `eval` during precompilation - don't do this.
Stacktrace:
 [1] eval
   @ ./boot.jl:370 [inlined]
 [2] eval(x::String)
   @ Artifacts ~/julia/usr/share/julia/stdlib/v1.9/Artifacts/src/Artifacts.jl:3
 [3] var"@artifact_str"(__source__::LineNumberNode, __module__::Module, name::Any, platform::Any, artifacts_toml_path::Any)
   @ Artifacts ~/julia/usr/share/julia/stdlib/v1.9/Artifacts/src/Artifacts.jl:665
```

Reply:
No, I only tried it with the test set, which doesn't hit this error, apparently.  @vtjnash what is the correct fix here?  I don't really understand how to properly "get" the value of a string at compile-time I guess.

Reply:
What was it that failed before this change? I tried it with JLLWrappers and it seemed to work ok without the `eval`.